esp/services/espcontrol/ws_espcontrolservice.cpp

```diff
@@ -51,3 +131,210 @@ bool CWSESPControlEx::onSetLogging(IEspContext& context, IEspSetLoggingRequest&
+
+        VStringBuffer xpath("/%s/%s[@name='%s']/%s[@name='%s']/%s[@port=\"%d\"]", PathSessionRoot, PathSessionProcess,
+            espProcess.get(), PathSessionDomain, authDomain.str(), PathSessionApplication, port);
+        Owned<IRemoteConnection> globalLock = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+        if (!globalLock)
+            throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Unable to connect to ESP Session information in dali %s", xpath.str());
+
+        IPropertyTree* espAuthDomainApp = globalLock->queryRoot();
+        if (!espAuthDomainApp)
+            throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Unable to get ESP Authentication Domain %s for port %u from dali", authDomain.str(), port);
+
+        if (sessionID && *sessionID)
+            xpath.setf("%s[%s='%s']", PathSessionSession, PropSessionID, sessionID.str());
+        else
+            xpath.setf("%s[%s='%s']", PathSessionSession, PropSessionNetworkAddress, fromIP.str());
+        Owned<IPropertyTreeIterator> iter = espAuthDomainApp->getElements(xpath.str());
+        ForEach(*iter)
+        {
+            IPropertyTree& item = iter->query();
+            espAuthDomainApp->removeTree(&item);
+        }
+
+        resp.setStatus(0);
+        resp.setMessage("Session is cleaned.");
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+bool CWSESPControlEx::onSetSessionTimeout(IEspContext& context, IEspSetSessionTimeoutRequest& req, IEspSetSessionTimeoutResponse& resp)
+{
+    try
+    {
+#ifdef _USE_OPENLDAP
+        CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
+        if(secmgr && !secmgr->isSuperUser(context.queryUser()))
+            throw MakeStringException(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Failed to set session timeout. Permission denied.");
+#endif
+
+        StringBuffer sessionID = req.getSessionID();
+        StringBuffer fromIP = req.getFromIP();
+        if (sessionID.trim().isEmpty() && fromIP.trim().isEmpty())
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "SessionID or FromIP has to be specified.");
+
+        StringBuffer authDomain = req.getAuthDomain();
+        if (authDomain.isEmpty())
+            authDomain.set("default");
+        unsigned port = 8010;
+        if (!req.getPort_isNull())
+            port = req.getPort();
+
+        VStringBuffer xpath("/%s/%s[@name='%s']/%s[@name='%s']/%s[@port=\"%d\"]", PathSessionRoot, PathSessionProcess,
+            espProcess.get(), PathSessionDomain, authDomain.str(), PathSessionApplication, port);
+        Owned<IRemoteConnection> globalLock = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+        if (!globalLock)
+            throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Unable to connect to ESP Session information in dali %s", xpath.str());
```

Comment:
See previous comment - looks very similar to other code, should be commoned up if at all possible.